src/domain/relay/limit-addresses.mapper.ts

```diff
@@ -109,4 +124,57 @@ export class LimitAddressesMapper {
 
     return firstRecipient;
   };
+
+  private isValidCreateProxyWithNonce(chainId: string, data: Hex): boolean {
+    const isCreateProxyWithNonce = this.proxyFactoryDecoder.isFunctionCall({
+      functionName: 'createProxyWithNonce',
+      data,
+    });
+
+    if (!isCreateProxyWithNonce) {
+      return false;
+    }
+
+    const decoded = this.proxyFactoryDecoder.decodeFunctionData({
+      data,
+    });
```

Comment:
We are doing repeated work here right? `isFunctionCall` ends up decoding the data. If that is successful, the decoded data is available. But here we decode the function data again.

Reply:
I adjusted it accordingly in 1eaf7b41dce41bde31c88132051d63b47789c8d2.